packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncTimeoutOptionsBuilder.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.mongodb.sync
+
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Builder for configuring various timeouts related to the sync connection to the server.
+ *
+ * @see [io.realm.kotlin.mongodb.AppConfiguration.Builder.syncTimeouts]
+ */
+public class SyncTimeoutOptionsBuilder {
+
+    /**
+     * The maximum time to allow for a connection to become fully established. This includes
+     * the time to resolve the network address, the TCP connect operation, the SSL
+     * handshake, and the WebSocket handshake.
+     *
+     * Only values >= 1 second is allowed. Default is 2 minutes.
+     *
+     * @throws IllegalArgumentException if the duration is outside the allowed range.
+     */
+    public var connectTimeout: Duration = 2.minutes
+        set(value) {
+            if (value < 1.seconds) {
```

Comment:
Maybe use `Validation.require(value < 1.seconds), { "connectionTimeout..."})` for consistency.

Reply:
Here and throughout the class.

Reply:
Done